### PR TITLE
more ifdef cleanup and removal

### PR DESCRIPTION
**Pull request by lrusak**
Further slims down `xbmc/system.h`

This moves all of the defines to cmake now. Some othe the defines may have odd names and stuff but this can be cleaned up in a future PR.

The only thing I'm not 100% sure about is using `SYSTEM_DEFINES`
```
list(APPEND SYSTEM_DEFINES -DHAS_LINUX_NETWORK)
```
It's just a name but I'm not sure if there is a better variable to choose or if we should make a new one.

I also noticed that the defines `HAS_EVENT_SERVER`, `HAS_JSONRPC`, and `HAS_FILESYSTEM_CDDA` are useless as there is no way to disable them. I removed them from the code entirely.

build and run tested on linux64 and linux64-gbm already.

### CODE REVIEW

File: xbmc/system.h [hunk 1]
```diff
@@ -159,25 +73,6 @@
...
-#ifdef HAVE_LIBGLESV3
-#undef HAS_GLES
-#define HAS_GLES 3
-#endif
-
-#ifdef HAS_DVD_DRIVE
-#define HAS_CDDA_RIPPER
-#endif
-
 #define SAFE_DELETE(p)       do { delete (p);     (p)=NULL; } while (0)
```
Comment (garbear) [inline xbmc/system.h]: These can be moved to another (possibly new) header

Comment (lrusak) [reply xbmc/system.h]: For a future PR ;)

**Comment by notspiff** [comment]
Better yet drop the macros altogether imo.

20. des. 2017 19:23 skrev "Garrett Brown" <notifications@github.com>:

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
@Rechi any idea about the window build error?
```
  C:\jenkins_slave\workspace\WIN-32\project\BuildDependencies\win32\include\cdio/read.h(67): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (compiling source file C:\jenkins_slave\workspace\WIN-32\xbmc\platform\win32\WIN32Util.cpp) [C:\jenkins_slave\workspace\WIN-32\kodi-build\libkodi.vcxproj]
  C:\jenkins_slave\workspace\WIN-32\project\BuildDependencies\win32\include\cdio/read.h(67): error C2146: syntax error: missing ';' before identifier 'cdio_read' (compiling source file C:\jenkins_slave\workspace\WIN-32\xbmc\platform\win32\WIN32Util.cpp) [C:\jenkins_slave\workspace\WIN-32\kodi-build\libkodi.vcxproj]
  c:\jenkins_slave\workspace\win-32\xbmc\platform\win32\PlatformDefs.h(36): error C2378: 'ssize_t': redefinition; symbol cannot be overloaded with a typedef (compiling source file C:\jenkins_slave\workspace\WIN-32\xbmc\platform\win32\WIN32Util.cpp) [C:\jenkins_slave\workspace\WIN-32\kodi-build\libkodi.vcxproj]
```

File: xbmc/storage/cdioSupport.h [hunk 2]
```diff
@@ -27,10 +27,6 @@
 
 #pragma once
 
-#include "system.h" // for HAS_DVD_DRIVE
```
Comment (Rechi) [inline xbmc/storage/cdioSupport.h]: @lrusak this is causing the windows build failure

File: xbmc/storage/cdioSupport.h [hunk 3]
```diff
@@ -29,8 +29,6 @@
 
 #include "system.h" // for HAS_DVD_DRIVE
```
Comment (Rechi) [inline xbmc/storage/cdioSupport.h]: The comment is now wrong, include is needed for `ssize_t` typedef.
I think including `PlatformDefs.h` would be enough.

Comment (lrusak) [reply xbmc/storage/cdioSupport.h]: let me try :D

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
squashed as jenkins was happy

**Comment by lrusak** [comment]
jenkins build this please

**Pull request merged**